### PR TITLE
Upgrade to Go 1.7

### PR DESCRIPTION
**Pull request by zimmski**

### CODE REVIEW

**Comment by zimmski** [comment]
This PR is ready, I will integrate -msan if i find a solution to #123

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: go
 go:
-  - 1.5
+  - 1.7
```
Comment (sbinet) [inline .travis.yml]: shouldn't this be `1.7.1` ?

Comment (zimmski) [inline .travis.yml]: You mean be explicit about the patch version? ("1.7" uses the latest 1.7 release)

Comment (zimmski) [inline .travis.yml]: OK it **should** use the latest patch release but it does not https://travis-ci.org/go-clang/gen/builds/158708830

Comment (zimmski) [inline .travis.yml]: done. I also added #124 to these PRs because I needed to change the CC env anyway.

**Comment by zimmski** [comment]
Finally, the "env" TravisCI configuration was driving me crazy.

**Comment by zimmski** [comment]
I still have not finished the address sanitizer issue but this PR is rebased and ready to go. Here is the corresponding bootstrap PR https://github.com/go-clang/bootstrap/pull/11

File: Makefile [hunk 2]
```diff
@@ -26,8 +30,10 @@ install-tools:
...
 	go get -u github.com/mattn/goveralls/...
 lint: install
-	scripts/lint.sh
+	$(ROOT_DIR)/scripts/lint.sh
 test:
-	CGO_LDFLAGS="-L`llvm-config --libdir`" go test -timeout 60s -race ./...
+	CGO_LDFLAGS="-L`llvm-config --libdir`" go test -timeout 60s ./...
+test-full:
+	$(ROOT_DIR)/scripts/test-full.sh
 test-verbose:
```
Comment (marxriedl) [inline Makefile]: same as in bootstrap, why do we have both, test-full, test-verbose?

Comment (zimmski) [inline Makefile]: Let's discuss that in the other PR https://github.com/go-clang/bootstrap/pull/11#r79457739

**Pull request merged**